Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/RCFCaster.java

```diff
@@ -246,6 +247,34 @@ public TimedRangeVector extrapolate(Calibration calibration, int horizon, boolea
 
     @Override
     public List<AnomalyDescriptor> processSequentially(double[][] data, Function<AnomalyDescriptor, Boolean> filter) {
+        if (data == null || data.length == 0) {
+            return new ArrayList<>();
+        }
+
+        long timestamp = preprocessor.getInternalTimeStamp();
+        long[] timestamps = new long[data.length];
+        for (int i = 0; i < data.length; i++) {
+            timestamps[i] = ++timestamp;
+        }
+
+        return processSequentially(data, timestamps, filter);
+    }
+
+    public List<AnomalyDescriptor> processSequentially(double[][] data, long[] timestamps, Function<AnomalyDescriptor, Boolean> filter) {
```

Comment:
nit: adding a filter is not null check in preconditions

Reply:
added